### PR TITLE
add symmetric tests on toJson where we had fromJson

### PR DESCRIPTION
**Pull request by bobagold**
Hi there!
I've run into the same problem as here https://github.com/lejard-h/google_maps_webservice/issues/75 and made a quick fix. I tried not to change much code. Please let me know what do you think.
Best,
Vladimir

### CODE REVIEW

**Comment by AAllport** [comment]
Hi @bobagold ,
I'm the guy who wrote #75 ,
Just wondering, is there any reason you added `var map = <String, dynamic>{};` as opposed to converting the functions to expressions, since I believe that would also solve the issue?

**Comment by bobagold** [comment]
Hi @AAllport , thank you for your feedback 👍  
I think that type-hinting of map literals is required by current dart version in both use cases: function expressions and function bodies. I just followed instructions of flutter analyzer.

**Comment by AAllport** [comment]
Ah fair enough, looks good though!

File: lib/src/geolocation.dart [hunk 1]
```diff
@@ -118,6 +118,15 @@ class GeolocationResponse extends GoogleResponseStatus {
       json.containsKey('error') ? json['error']['message'] : null,
       Location.fromJson(json['location']),
       json['accuracy']);
+
+  @override
+  Map<String, dynamic> toJson() {
+//    var map = super.toJson();
```
Comment (lejard-h) [inline lib/src/geolocation.dart:124]: you can remove commented line :)

and explain why you don't call `super.toJson()`

File: lib/src/core.dart [hunk 2]
```diff
@@ -51,11 +51,17 @@ class Geometry {
       : null;
 
   Map<String, dynamic> toJson() {
-    var map = {};
+    var map = <String, dynamic>{};
     map['location'] = (location != null) ? location.toJson() : null;
-    map['location_type'] = locationType;
-    map['viewport'] = (viewport != null) ? viewport.toJson() : null;
-    map['bounds'] = (bounds != null) ? bounds.toJson() : null;
+    if (locationType != null) {
```
Comment (lejard-h) [inline lib/src/core.dart:56]: maybe we should do the same for every `toJson` methods and avoid sending null value

Comment (lejard-h) [reply lib/src/core.dart:56]: Can you at least do it for every `toJson` method you modified.

We can patch other `toJson` method in a second PR

Comment (lejard-h) [reply lib/src/core.dart:56]: Or maybe it's time to use json_serializable ...

**Comment by AAllport** [comment]
Point of discussion: @lejard-h, is there any reason not to migrate to json_serializable (or similar)?

**Comment by lejard-h** [comment]
Not really, when I started the package, json_serializable was new and not really stable, so it was easier to do it manually.

Now it's getting harder to add new feature and harder to read, so yes it would be a great improvement

**Comment by AAllport** [comment]
@bobagold, are you going to add `json_serializable` or did you want someone else to contribute?

**Comment by lejard-h** [comment]
Thanks for your PR but `json_serializable` along with null-safety support is now on `master` branch
I can close this

**Pull request closed without merging**